### ISSUE TITLE
Make timeline available with x scales other than date

### ISSUE THREAD

**Issue by thetravis** [swimlane/ngx-charts#624]
**I'm submitting a ...**  (check one with "x")
```
[x] feature request
```

**Current behavior**

According to documentation `timeline` is 
> Only available if x scale is date

**Expected behavior**

Timeline should work with x scale in numbers (linear/time?).

**Reproduction of the problem**

Set `[timeline]="true"` and
>  data = [{
>    "name": "Germany",
>    "series": [
>      {
>        "name": 2010,
>        "value": 7300000
>      },
>      {
>        "name": 2011,
>        "value": 8940000
>      },
>      {
>        "name": "2015",
>        "value": 8960000
>      },
>    ]
>  }];

and timeline doesn't show.  

**What is the motivation / use case for changing the behavior?**

I would like to use timeline with my data, but since the `name` values in my data are not dates but numbers (milliseconds in this case), it doesn't work.

**Comment by sandangel** [comment]
you can do it, just copy the source graph, create a new custom graph, at the timeline prop, remove condition and set it to true.

**Comment by thetravis** [comment]
Why is it limited to dates only? I'm wondering if I'm doing something wrong. For me at least it makes sense to have timeline with other data aswell if you can arrange them.

**Comment by sandangel** [comment]
I don't think this library provides ready-to-use all kind of charts. If you need somethings else, build your own. The only limit here is yourself.

**Comment by thetravis** [comment]
I finally got it working! 

I copied the original `line-chart.component.ts` into my project and rename it to `line-chart-timeline.component.ts` and the selector to `ngx-charts-line-chart-timeline`. I corrected the paths in `import`s to point to `../../node_modules/@swimlane/ngx-charts/release/` and copied the `base-chart.css` to my project aswell and fixed the `styleUrls` reference to it. In `AppModule` I imported the new `LineChartTimelineComponent` and added it to `declarations`

I also had to change `includes` to `indexOf` since my compiler gave me an error `'includes' does not exist on type 'any[]'`.
New version of getXDomain:
>  getXDomain(): any[] {
>    let values = [];
>
>    for (const results of this.results) {
>      for (const d of results.series){
>        if (values.indexOf(d.name) == -1) { // used to be !values.includes(d.name)
>          values.push(d.name);
>        }
>      }
>    } 
>    ...

and got it to compile.

In the template I removed `scaleType == 'time'` clause

> <svg:g ngx-charts-timeline
>        *ngIf="timeline"

And now it shows the timeline :)

So my question is: why is that scaleType == 'time' in there dy default?

**Comment by marjan-georgiev** [comment]
@thetravis the timeline does not work on ordinal scales, but it can work on linear scales (as is your case). Our use case for it was only for dates, which is why we named the feature a timeline. 

We can consider adding support for linear scales.

**Comment by thetravis** [comment]
Would ` *ngIf="scaleType != 'ordinal'"` do the trick?

**Comment by marjan-georgiev** [comment]
Yup. If you're willing to do a PR and test that, please go ahead.

**Comment by thetravis** [comment]
I'm just trying to make my first pull request to another project. Let's see once I get it done. 

Also, I have another problem with timeline. I have function that updates the data once per sec. If I select something in the timeline and the click again to clear the selection the new values won't show up in the graph. (The graph stops 'rolling'.) I'll open a new issue for it later.

**Comment by thetravis** [comment]
I made a fork, edited the files and ran `npm run release` but wasn't able to include it in my project, possibly because I'm using Angular 4 with old version of ngx-graphs and the fork uses Angular 5. 

Using `*ngIf="timeline && scaleType != 'ordinal'"` in the template seems to work with my data, so I copied it to area charts and line chart where I could find `*ngIf="timeline ...` 

At the moment I have no time to test it more. If it doesn't work with yours you always have option to use `timeline=false`. 

I'll make pull request.
